src/common/helpers/dynamics/dynamics-client.js

```diff
@@ -6,6 +6,125 @@ import querystring from 'node:querystring'
 import { StatusCodes } from 'http-status-codes'
 import { REQUEST_QUEUE_STATUS } from '../../constants/request-queue.js'
 import { isOrganisationEmployee } from '../organisations.js'
+import { createLogger } from '../../helpers/logging/logger.js'
+
+const logger = createLogger()
+
+/**
+ * Fetches an exemption from the database by application reference
+ * @param {Object} db - Database connection
+ * @param {string} applicationReferenceNumber - Application reference number
+ * @returns {Promise<Object>} The exemption document
+ */
+const fetchExemption = async (db, applicationReferenceNumber) => {
+  const exemption = await db.collection('exemptions').findOne({
+    applicationReference: applicationReferenceNumber
+  })
+
+  if (!exemption) {
+    throw Boom.notFound(
+      `Exemption not found for applicationReference: ${applicationReferenceNumber}`
+    )
+  }
+
+  return exemption
+}
+
+/**
+ * Updates the queue status to IN_PROGRESS
+ * @param {Object} db - Database connection
+ * @param {string} exemptionId - Exemption ID
+ */
+const updateQueueStatus = async (db, exemptionId) => {
+  await db.collection('exemption-dynamics-queue').updateOne(
+    { _id: exemptionId },
+    {
+      $set: {
+        status: REQUEST_QUEUE_STATUS.IN_PROGRESS,
+        updatedAt: new Date()
+      }
+    }
+  )
+}
+
+/**
+ * Builds the Dynamics API payload
+ * @param {Object} exemption - The exemption document
+ * @param {string} applicationReferenceNumber - Application reference number
+ * @param {string} frontEndBaseUrl - Frontend base URL
+ * @returns {Object} The payload for Dynamics API
+ */
+const buildDynamicsPayload = (
+  exemption,
+  applicationReferenceNumber,
+  frontEndBaseUrl
+) => {
+  const organisationId = exemption.organisation?.id
+  const beneficiaryOrganisationId = organisationId
+  const applicantOrganisationId = isOrganisationEmployee(exemption.organisation)
+    ? organisationId
+    : undefined
+
+  return {
+    contactid: exemption.contactId,
+    projectName: exemption.projectName,
+    reference: applicationReferenceNumber,
+    type: EXEMPTION_TYPE.EXEMPT_ACTIVITY,
+    applicationUrl: `${frontEndBaseUrl}/view-details/${exemption._id}`,
+    ...(applicantOrganisationId ? { applicantOrganisationId } : {}),
```

Comment:
Should we swap the names of these two over as we're using the opposite meanings I think?